plugins/bulk-import/src/components/PreviewFile/PreviewFile.tsx

```diff
@@ -80,55 +90,111 @@ export const PreviewFileSidebar = ({
   onClose,
   repositoryType,
   data,
-  formErrors,
-  setFormErrors,
   handleSave,
   isSubmitting,
 }: {
   open: boolean;
   data: AddRepositoryData;
   repositoryType: RepositoryType;
   onClose: () => void;
-  formErrors: PullRequestPreviewData;
   isSubmitting?: boolean;
-  setFormErrors: (formErr: PullRequestPreviewData) => void;
   handleSave: (pullRequest: PullRequestPreviewData, _event: any) => void;
 }) => {
   const classes = useDrawerStyles();
+  const bulkImportApi = useApi(bulkImportApiRef);
+  const [formErrors, setFormErrors] = React.useState<PullRequestPreviewData>();
+  const [isLoading, setIsLoading] = React.useState(true);
   const [pullRequest, setPullRequest] = React.useState<PullRequestPreviewData>(
     {},
   );
   const [isInitialized, setIsInitialized] = React.useState(false);
 
   const contentClasses = useDrawerContentStyles();
-  const initializePullRequest = React.useCallback(() => {
+  const initializePullRequest = React.useCallback(async () => {
     const newPullRequestData: PullRequestPreviewData = {};
 
     if (Object.keys(data?.selectedRepositories || [])?.length > 0) {
-      Object.values(data?.selectedRepositories || []).forEach(repo => {
-        if (repo.catalogInfoYaml?.prTemplate) {
-          newPullRequestData[repo.id] = repo.catalogInfoYaml.prTemplate;
+      setIsLoading(true);
+      Object.values(data?.selectedRepositories || []).forEach(async repo => {
+        const result = await bulkImportApi.getImportAction(
```

Comment:
async await might not work as expected with forEach

https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop

Reply:
You are right, this was overlooked. Thank you

Fixed it now